vizro-ai/examples/dashboard_ui/components.py

```diff
@@ -0,0 +1,214 @@
+"""Contains custom components used within a dashboard."""
+
+try:
+    from pydantic.v1 import Field, validator
+except ImportError:
+    from pydantic import Field
+
+from typing import List, Literal
+
+import dash_bootstrap_components as dbc
+import vizro.models as vm
+from dash import dcc, html
+from pydantic import PrivateAttr
+from vizro.models import Action
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._models_utils import _log_call
+
+try:
+    from pydantic.v1 import Field
+except ImportError:  # pragma: no cov
+    pass
+
+import black
+
+
+class UserPromptTextArea(vm.VizroBaseModel):
+    """Input component `UserPromptTextArea`.
+
+    Based on the underlying [`dcc.Input`](https://dash.plotly.com/dash-core-components/input).
+
+    Args:
+        type (Literal["user_input"]): Defaults to `"user_text_area"`.
+        title (str): Title to be displayed. Defaults to `""`.
+        placeholder (str): Default text to display in input field. Defaults to `""`.
+        actions (Optional[List[Action]]): Defaults to `[]`.
+
+    """
+
+    type: Literal["user_text_area"] = "user_text_area"
+    actions: List[Action] = []
+
+    _set_actions = _action_validator_factory("value")
+
+    @_log_call
+    def build(self):
+        """Returns the text area component to display vizro-ai code output."""
+        return html.Div(
+            children=[
+                dcc.Textarea(
+                    id=self.id,
+                    placeholder="How can Vizro-AI help you today?",
```

Comment:
Maybe be more specific, so it's obvious to users that this UI is assisting on chart creation, and user questions should be focus on plotting and the data uploaded